sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventService.java

```diff
@@ -194,6 +194,31 @@ public EventService() {
 		return em.createQuery(cq).getResultList().stream().collect(Collectors.toMap(objects -> (String) objects[0], objects -> (User) objects[1]));
 	}
 
+	public Map<String, User> getAllEventUuidsWithResponsibleUserByPersonAndDiseaseAfterDateForNotification(String personUuid, String excludedEventUuid, Disease disease, Date date) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Object[]> cq = cb.createQuery(Object[].class);
+		Root<Event> eventRoot = cq.from(Event.class);
+		Join<Event, EventParticipant> eventParticipantJoin = eventRoot.join(Event.EVENT_PERSONS, JoinType.INNER);
+		Join<EventParticipant, Person> personJoin = eventParticipantJoin.join(EventParticipant.PERSON, JoinType.INNER);
+		Join<Event, User> responsibleUserJoin = eventRoot.join(Event.RESPONSIBLE_USER, JoinType.INNER);
+
+		Timestamp timestamp = DateHelper.toTimestampUpper(date);
+		Predicate filter = cb.and(
+			cb.equal(personJoin.get(Person.UUID), personUuid),
+			cb.not(cb.equal(eventRoot.get(Event.UUID), excludedEventUuid)),
+			cb.equal(eventRoot.get(Event.DISEASE), disease),
+			createActiveEventsFilter(cb, eventRoot),
+			cb.or(
+				CriteriaBuilderHelper.greaterThanAndNotNull(cb, eventRoot.get(Event.START_DATE), timestamp),
```

Comment:
are you certain this is what is required here?
from the ticket: 'reported in a space time of 30 days' - to me sounds like report date was in the last 30 days